README.md

```diff
@@ -117,28 +151,119 @@ In most cases, you should use a specification for metadata instead.
-    - name: set example vegetables
+          - user:/tests/ansible:
+              fruit:
+                cherry:
+                  - value: cola
+                  - meta:
+                      # This adds the type metakey with value string to the key /tests/ansible/fruit/cherry
+                      type: string
+                  - keys:
+                      pie: yummy
+                      jelly:
+                        sugar: none           
+            pets:
+              - array:
+                  - name: rufus
+                    type: dog
+                  - name: bessie
+                    type: cow
+            friends:
+              - meta:
+                  complete: false
+              - array:
+                  - '#': Mary
+                  - '#':
+                    - value: John
+                    - meta:
+                        age: 53
+```
+
+## Session Recording
+
+You can control Elektra's session recording mechanism with this module.
+The `recording` element has the following parameters:
+- `enable`: `true` or `false`. 
+            Whether session recording should be enabled after the task is complete.
+            By default, we enable this.
+- `parentKey`: The parent key to use for session recording.
+               Every change to this key or below will be recorded.
+- `clear`: `true` or `false` 
+            Whether the current recording session should be deleted.
+            By default, we clear it.
+- `recordAnsible`: `true` or `false`.
+                   Whether changes made to Elektra via this task should be recorded.
+                   By default, we do not do this.
+
+```yml
+- name: elektra module example
+  hosts: localhost
+  connection: local
+  collections:
+    - elektra_initiative.libelektra
+  tasks:
+    - name: set example keys
       elektra:
-        mountpoint: user:/tests/ansible/vegetables
-        keys:
-          tomato:
-              value: ketchup
-          potato:
-              value: fries
+        recording:
+          enable: true
+          clear: true
+          parentKey: user:/myapp
+          recordAnsible: true
+```
+
+## Merging
+
+You can force a 3-way merge to be used to merge the keys with existing changes on a host.
+The `merge` element has the following parameters:
+- `strategy`: The strategy to use to resolve merge conflicts. Either one of:
+  - `ours`: use the keys specified in this tasks on conflict.
+  - `theirs`: use the keys on the host on conflict.
+  - `abort`: abort task on conflict.
+- `base`: If specified, this is the base keyset to use as the _base_ in the 3-way merge.
+          It has the same syntax as the `key` element of the task.
+          If not specified, we will try to generate the base keys from the current configuration on the host.
+          If the recording session on the host contains changes, we will undo them from the base.
+
+## Other Options
+
+`clear`: A list of keys that shall be removed, including their children.
```

Comment:
I think recursive also should be a flag, you might want to remove individual keys.
Terminology inconsistent (please stick to Elektra's terms): 

```suggestion
`remove`: A list of keys that shall be removed, including their children.
```

Reply:
done